src/core/Directus/Application/CoreServicesProvider.php

```diff
@@ -344,9 +343,9 @@ protected function getEmitter()
 
                 $payload->replace($recordData);
                 $payload->remove('data');
+                $payload->remove('filename');
                 $payload->remove('html');
 
-                $payload->set('id', $data['id']);
                 $payload->set('private_hash', get_random_string());
```

Comment:
It doesn't feel right to me that this `/asset/` endpoint specific code is added in so deep down the abstraction layer in CoreServicesProvider. Lets move that to AssetsService

Reply:
This is the older structure - in which the hooks are defined to save file action. May I know what exactly you are expecting to be changed?

Reply:
May I have a brief on what should be replaced? As this hook is predefined [Which execute the specific files operation on item.update and item.create event] - PR just updating the code based on our new requirement.